.github/workflows/integration_tests.yml

```
@@ -87,10 +49,31 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+            artifacts/
+          key: ${{ runner.os }}-cargo-with-artifacts-${{ hashFiles('**/Cargo.lock') }}
+
+      # WARNING: if we dont `mkdir -p artifacts target` before we optimize the contracts
+      # then we will get os permission errors trying to later build the workspace on a cache miss,
+      # because the workspace-optimizer docker container is creating those folders with incorrect perms
+      # if they dont already exist.
+      - name: Download smart contract dependencies
+        run: |
+          mkdir -p artifacts target
+          wget https://github.com/CosmWasm/cw-plus/releases/download/v0.11.1/cw20_base.wasm \
+            -O artifacts/cw20_base.wasm
+          wget https://github.com/CosmWasm/cw-plus/releases/download/v0.11.1/cw4_group.wasm \
+            -O artifacts/cw4_group.wasm
```

```suggestion
          wget https://github.com/CosmWasm/cw-plus/releases/latest/download/cw20_base.wasm \
            -O artifacts/cw20_base.wasm
          wget https://github.com/CosmWasm/cw-plus/releases/latest/download/cw4_group.wasm \
            -O artifacts/cw4_group.wasm
```

From the bottom of this page: https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases

Probably good to test against the latest stuff. 🤷‍♂️

---

Great catch, will do.